jwst/regtest/st_fitsdiff.py

```diff
@@ -0,0 +1,777 @@
+            (default: 0.0). Values which satisfy the expression
+
+            .. math::
+
+                \\left| a - b \\right| > \\text{atol} + \\text{rtol} \\cdot \\left| b \\right|
+
+            are considered to be different.
+            The underlying function used for comparison is `numpy.allclose`.
+
+            .. versionadded:: 2.0
+
+        atol : float, optional
+            The allowed absolute difference. See also ``rtol`` parameter.
+
+            .. versionadded:: 2.0
+
+        ignore_blanks : bool, optional
+            Ignore extra whitespace at the end of string values either in
+            headers or data. Extra leading whitespace is not ignored
+            (default: True).
+
+        ignore_blank_cards : bool, optional
+            Ignore all cards that are blank, i.e. they only contain
+            whitespace (default: True).
+
+        report_pixel_loc_diffs : bool, optional
+            Report all the pixel locations where differences are found.
+        """
+        self.report_pixel_loc_diffs = report_pixel_loc_diffs
+        self.nans = []
+        self.percentages = {}
+        self.stats = {}
+        super().__init__(a, b,
+                         ignore_keywords=ignore_keywords,
+                         ignore_comments=ignore_comments,
+                         ignore_fields=ignore_fields,
+                         numdiffs=numdiffs,
+                         rtol=rtol,
+                         atol=atol,
+                         ignore_blanks=ignore_blanks,
+                         ignore_blank_cards=ignore_blank_cards)
+
+    def _diff(self):
+        if self.a.name != self.b.name:
+            self.diff_extnames = (self.a.name, self.b.name)
+
+        if self.a.ver != self.b.ver:
+            self.diff_extvers = (self.a.ver, self.b.ver)
+
+        if self.a.level != self.b.level:
+            self.diff_extlevels = (self.a.level, self.b.level)
+
+        if self.a.header.get("XTENSION") != self.b.header.get("XTENSION"):
+            self.diff_extension_types = (
+                self.a.header.get("XTENSION"),
+                self.b.header.get("XTENSION"),
+            )
+
+        self.diff_headers = STHeaderDiff.fromdiff(
+            self, self.a.header.copy(), self.b.header.copy()
+        )
+
+        def get_percentage(values, threshold):
+            # Calculate percentage
+            n_total = values.size
+            n = np.logical_or(values > threshold, values < -threshold).nonzero()[0].size
+            return np.round((n / n_total) * 100, decimals=2)
+
+        def get_quick_report(a, b):
+            # match nans for all arrays and remove them for logical comparison
+            nan_idx = (np.isnan(a) | np.isnan(b))
+            anonan = a[~nan_idx]
+            bnonan = b[~nan_idx]
+            # Get the number of NaN in each array
+            nans = [np.isnan(a).size, np.isnan(b).size]
+            # Calculate stats
+            values = np.abs(np.abs(anonan) - np.abs(bnonan))
```

Comment:
Do you want to take the absolute values of a and b before taking the difference?  This would make 4 .0 and -4.0 equal

Reply:
Ah! good catch! yeah, nope. Definitely an oops.